app/lib/package.rb

```diff
@@ -52,6 +55,15 @@ def determine_resources(document)
     nodes.each { |node| @resources << node.content }
   end
 
+  def determine_tips(document)
+    nodes = XmlUtil.xpath_namespace(document, XPATH_TIPS.join("|"))
+    nodes.each do |node| 
+      tip = @translation.resource.tips.find_by(name: node.content)
+      raise ActiveRecord::RecordNotFound, "Tip not found: #{node.content}" unless tip
```

Comment:
because `add_tips` should be gated on `include_tips`, you will probably want to just track the tip names in `@tips` and raise missing tips within the `add_tips` logic

Reply:
Right, I did it just a bit differently but am checking the flag now